dfaker.py

```diff
@@ -0,0 +1,387 @@
+				sys.exit(1)
+		except:
+			print('Wrong time format: {:s}. Please enter time as HH:MM'.format(args.time))
+			sys.exit(1)
+	 
+	if args.zone:
+		if args.zone not in pytz.common_timezones:
+			print('Unrecongznied zone: {:s}'.format(args.zone))
+			sys.exit(1)
+		params['zone'] = args.zone
+
+	if args.num_days:
+		params['num_days'] = args.num_days
+
+	if args.file:
+		if args.file[-5:] != '.json':
+			print('Output file must be in be a json file')
+			sys.exit(1)
+		params['file'] = args.file
+
+	if args.minify:
+		params['minify'] = True
+
+	if args.gaps:
+		params['gaps'] = True
+
+	if args.smbg_freq:
+		if smbg_freq == 'high':
+			params['smbg_freq'] = 8
+		elif smbg_freq == 'average':
+			params['smbg_freq'] = 6
+		elif smbg_freq == 'low':
+			params['smbg_freq'] = 3
+		else:
+			print('Invalid frequency: {:s}. Please choose between high, average and low'.format(args.smbg_freq))
+			sys.exit(1)
+
+parser = argparse.ArgumentParser()
+parser.add_argument('-tz', '--timezone', dest='zone', help='Local timezone')
+parser.add_argument('-d', '--date', dest='date', help='Date in the following format: YYYY-MM-DD')
+parser.add_argument('-t', '--time', dest='time', help='Time in the following format: HH:MM')
+parser.add_argument('-nd', '--num_days', dest='num_days',
+					help='Number of days to generate data. Default = 180')
+parser.add_argument('-f', '--output_file', dest='file', help='Name of output json file')
+parser.add_argument('-m', '--minify', dest='minify', action='store_true', help='Minify the json file')
+parser.add_argument('-g', '--gaps', dest='gaps', action='store_true', help='Add gaps to fake data')
+parser.add_argument('-s', '--smbg', dest='smbg_freq', help='Freqency of fingersticks a day: high, average or low')
+args = parser.parse_args()
+parse(args, params)
+
+def apply_loess(params, solution):
+	"""Solves the blood glucose equation over specified period of days 
+		and applies a loess smoothing regression to the data 
+		Returns a numpy array of glucose time values 
+	"""
+	#solving blood glucose eqn 
+	glucose = solution[:, 1]
+	time = solution[:, 2]
+	#smoothing blood glucose eqn
+	lowess = sm.nonparametric.lowess
+	smoothing_distance = 1.3 #1.3 minutes
+	fraction = (smoothing_distance / (params['num_days'] * 60 * 24)) * 100
+	result = lowess(glucose, time, frac=fraction, is_sorted=True)
+	
+	if params['gaps']:
+		result = make_gaps(params, result)
+	smoothed_time = result[:, 0]
+	smoothed_glucose = result[:, 1]
+	return smoothed_glucose, smoothed_time
+
+def get_offset(params):
+	utc_tz = timezone('UTC')
+	local_tz = timezone(params['zone'])
+	naive = datetime(params['year'], params['month'], params['day'], params['hour'], params['minute'])
+	offset = int((utc_tz.localize(naive) - local_tz.localize(naive)) / timedelta(minutes=1))
+	return offset
+
+def make_gaps(params, time_gluc):
+	gaps = random.randint(4 * params['num_days'], 5 * params['num_days']) # amount of gaps	
```

Comment:
I was trying to play with the numbers, originally they were 1 and 3 so that on average there would be between 1 and 3 gaps a day -- I don't remember why I changed it to 4/5, probably to see if it made a significant difference, but i've changed it back to 1 and 3 now.